vignettes/cancensus.Rmd

```diff
@@ -0,0 +1,195 @@
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Vignette Title}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, echo=FALSE, message=FALSE, warning=FALSE}
+knitr::opts_chunk$set(include = TRUE)
+library(cancensus)
+library(dplyr)
+# options(cancensus.api_key = "your_api_key")
+```
+# Canadian Census Data
+
+Statistics Canada conducts the Census of the Canadian population every five years, with the most recent Census completed in 2016. Statistics Canada [provides](http://www12.statcan.gc.ca/datasets/index-eng.cfm?Temporal=2016) datasets for Census products which are available for the 2016 Census, the 2011 National Household Survey, as well as the 2006, 2001, 1996, and 1991 Censuses at different levels of geography. Statistics Canada provides this data in HTML, XML, and Beyond 20/20 formats.
+
+At this time, Statistics Canada does not provide an API service for Census data nor any other way to programmatically access Census data for reproducible research. 
+
+# Cancensus and CensusMapper
+
+The **cancensus** package was developed to provide users with a way to access Canadian Census in a programmatic way following good [tidy data](http://vita.had.co.nz/papers/tidy-data.pdf) practices. While the structure and data in **cancensus** is unique to Canadian Census data, this package is inspired in part by Kyle Walker's work on [tidycensus](https://github.com/walkerke/tidycensus), a package to interface with the US Census Bureau data APIs. 
+
+As Statistics Canada does not provide API access to Census data, **cancensus** retrieves Census data indirectly through the [CensusMapper](https://censusmapper.ca) API. CensusMapper is a project by [Jens von Bergmann](https://twitter.com/vb_jens), one of the authors of **cancensus**, to provide interactive geographic visualizations of Canadian Census data. CensusMapper databases store all publically available data from Statistics Canada for the 2006, 2011, and 2016 Censuses. Censusmapper data can be accessed via an API and **cancensus** is built to interface directly with it. 
+
+## API Key
+
+**cancensus** requires a valid CensusMapper API key to use. You can obtain a free API key by [signing up](https://censusmapper.ca/users/sign_up) for a CensusMapper account. CensusMapper API keys are free; however, API requests are limited in volume. For larger quotas, please get in touch with Jens [directly](mailto:jens@censusmapper.ca).  
+
+To check your API key, just go to "Edit Profile" (in the top-right of the CensusMapper menu bar). Once you have your key, you can store it in your system environment so it is automatically used in API calls. To do so just enter `options(cancensus.api_key = "your_api_key")`.
+
+## Installing cancensus
+
+```{r load package, echo=TRUE, message=FALSE, warning=FALSE, eval = FALSE}
+# install.packages("devtools")
+devtools::install_github("mountainmath/cancensus")
+
+library(cancensus)
+options(cancensus.api_key = "your_api_key")
```

Comment:
You should move the `options()` call to an `eval=FALSE` block at the end of the previous section, instead of here., since it seems kind of redundant and should be with the text that explains what the call is for. Just have the install info in this block -- nix the `library()` and `options()` calls. And there's no need for message/warning knitr settings on a block that isn't evaluation.

Reply:
Actually, now that I think about it: is there a reason to have an "installation" section in a vignette at all?

Reply:
I think so. With HTML vignettes and having static documentation online, you can point to the vignette for those who haven't installed the package yet. It may be redundant when combined with the homepage (Readme), but it lets it stand alone... I'm open to cutting it out, though as I don't think it's crucial.

Reply:
That's a fair point, and I'm not opposed to its inclusion.